### PR TITLE
feat(consumer): allow group instance id to enable static membership

### PR DESCRIPTION
**Pull request by dbanda**
We would like to experiment with running Kafka with static membership. This mode is enabled when consumer passing in the `group.instance.id` parameter.  This PR adds passing in the group instance id as a CLI parameter.

### CODE REVIEW

File: rust_snuba/src/consumer.rs [hunk 1]
```diff
@@ -215,6 +215,10 @@ pub fn consumer_impl(
         })
         .collect();
 
+    if let Some(group_instance_id) = consumer_config.group_instance_id {
+        broker_config.insert("group.instance.id".to_owned(), group_instance_id);
+    }
+
```
Comment (lynnagara) [inline rust_snuba/src/consumer.rs:221]: Can you have the `resolve_consumer_config()` function handle this? It's supposed to be the singular place where consumer config is resolved before passing it to the Rust or Python consumer so we can avoid needing to do stuff like this twice in Rust and Python

**Review by lynnagara — CHANGES_REQUESTED** [review]

File: rust_snuba/src/consumer.rs (hunk 1 above)
Comment (dbanda) [reply rust_snuba/src/consumer.rs:221]: I can do it on the python side, but I still need to a way to get that into `KafkaConfig::new_consumer_config`? Should I make `group_instance_id` a parameter to that function? Here I resolved to just adding it into the overrides hashmap.

File: rust_snuba/rust_arroyo/src/backends/kafka/config.rs [hunk 2]
```diff
@@ -85,6 +90,10 @@ mod tests {
                 "queued.max.messages.kbytes".to_string(),
                 "1000000".to_string(),
             )])),
+            Some(HashMap::from([(
+                "max.poll.interval.ms".to_string(),
+                "1000000".to_string(),
```
Comment (lynnagara) [inline rust_snuba/rust_arroyo/src/backends/kafka/config.rs:95]: What's this for?

Comment (dbanda) [reply rust_snuba/rust_arroyo/src/backends/kafka/config.rs:95]: It's a unit test. Refactored it to allow passing in key value pairs to the kafka config from the resolver.

File: rust_snuba/rust_arroyo/src/backends/kafka/config.rs [hunk 3]
```diff
@@ -23,6 +24,7 @@ impl KafkaConfig {
         auto_offset_reset: String,
         _strict_offset_reset: bool, // TODO: Implement this
         override_params: Option<HashMap<String, String>>,
+        consumer_config_params: Option<HashMap<String, String>>,
```
Comment (lynnagara) [inline rust_snuba/rust_arroyo/src/backends/kafka/config.rs:27]: This argument should not be necessary - there is already a way to pass consumer consumer config via `override_params` in the argument above this

File: snuba/consumers/consumer_config.py [hunk 4]
```diff
@@ -170,6 +175,7 @@ def resolve_consumer_config(
         max_batch_size=max_batch_size,
         max_batch_time_ms=max_batch_time_ms,
         env=resolved_env_config,
+        kafka_consumer_config=kafka_consumer_config,
```
Comment (lynnagara) [inline snuba/consumers/consumer_config.py:178]: Please don't add a new item here, Kafka consumer config should be part of the existing`raw_topic` configuration item

**Review by lynnagara — APPROVED** [review]
thanks!

**Pull request merged**